generator/src/main/java/com/scottlogic/deg/generator/generation/grouped/FieldSpecGroupValueGenerator.java

```diff
@@ -17,172 +17,177 @@
+        FieldSpecGroup newGroup = adjustBounds(generatedField, dataBag.getDataBagValue(generatedField), group);
 
-    private Stream<DataBag> createRemainingDataBags(Stream<DataBag> stream, Field field, FieldSpecGroup group) {
-        Stream<DataBagGroupWrapper> initial = stream
-            .map(dataBag -> new DataBagGroupWrapper(dataBag, group, underlyingGenerator))
-            .map(wrapper -> adjustWrapperBounds(wrapper, field));
-        Set<Field> toProcess = filterFromSet(group.fieldSpecs().keySet(), field);
-
-        Stream<DataBagGroupWrapper> wrappedStream = recursiveMap(initial, toProcess);
+        Stream<DataBag> dataBagStream = generate(newGroup)
+            .map(otherData -> DataBag.merge(dataBag, otherData));
 
-        return wrappedStream.map(DataBagGroupWrapper::dataBag);
+        return applyCombinationStrategy(dataBagStream);
     }
 
-    private static DataBagGroupWrapper adjustWrapperBounds(DataBagGroupWrapper wrapper, Field field) {
-        DataBagValue value = wrapper.dataBag().getDataBagValue(field);
-        FieldSpecGroup newGroup = adjustBounds(field, value, wrapper.group());
-        return new DataBagGroupWrapper(wrapper.dataBag(), newGroup, wrapper.generator());
+    private FieldSpecGroup adjustBounds(Field generatedField, DataBagValue generatedValue, FieldSpecGroup group) {
+        if (generatedValue.getValue() instanceof OffsetDateTime) {
 
-    }
+            Limit<OffsetDateTime> limit = new Limit<>((OffsetDateTime)generatedValue.getValue(), true);
+            LinearRestrictions<OffsetDateTime> restrictions = LinearRestrictionsFactory.createDateTimeRestrictions(limit,limit);
+            FieldSpec newSpec = FieldSpec.fromRestriction(restrictions).withNotNull();
 
-    private static Stream<DataBagGroupWrapper> recursiveMap(Stream<DataBagGroupWrapper> wrapperStream,
-                                                            Set<Field> fieldsToProcess) {
-        if (fieldsToProcess.isEmpty()) {
-            return wrapperStream;
+            return adjustBoundsOfDate(generatedField, newSpec, group);
         }
-
-        Field field = SetUtils.firstIteratorElement(fieldsToProcess);
-
-        Stream<DataBagGroupWrapper> mappedStream =
-            FlatMappingSpliterator.flatMap(wrapperStream, wrapper -> acceptNextValue(wrapper, field));
-
-        Set<Field> remainingFields = filterFromSet(fieldsToProcess, field);
-
-        return recursiveMap(mappedStream, remainingFields);
+        return group;
     }
 
-    private static <T> Set<T> filterFromSet(Set<T> original, T element) {
-        return original.stream()
-            .filter(f -> !f.equals(element))
+    private FieldSpecGroup adjustBoundsOfDate(Field generatedField, FieldSpec generatedValue, FieldSpecGroup group) {
+        Set<FieldSpecRelations> nonUpdatedRelations = group.relations().stream()
+            .filter(relation -> !relation.main().equals(generatedField) && !relation.other().equals(generatedField))
```

Comment:
Feels like following should be encapsulate din a method:
relation.main().equals(generatedField) || relation.other().equals(generatedField)
Can then use it when creating uipdateableRelatiosn and can use the negative of it when create nonUpdatedRelations.  could add a conatinsField(field) to relation class and thsi could implement that boolean check

Reply:
i've literally already made a method for this, i just forgot to use it 🤦‍♂

Reply:
ha ha.  cool